src/java/org/apache/cassandra/repair/autorepair/AutoRepair.java

```diff
@@ -179,6 +187,13 @@ public void repair(AutoRepairConfig.RepairType repairType)
 
             //consistency level to use for local query
             UUID myId = StorageService.instance.getHostIdForEndpoint(FBUtilities.getBroadcastAddressAndPort());
+
+            // If it's too soon to run repair, don't bother checking if it's our turn.
+            if (tooSoonToRunRepair(repairType, repairState, config, myId))
```

Comment:
Notable change:  We now check whether it's too soon to run repair before evaluating turn.  I figure this was for the best as it avoids querying auto repair history table and checking if replicas were running.